src/main/java/com/google/api/codegen/config/ResourceNameType.java

```diff
@@ -15,9 +15,18 @@
 package com.google.api.codegen.config;
 
 public enum ResourceNameType {
+  /** No resource type is available. */
   NONE,
+
+  /** A single concrete resource type. */
   SINGLE,
+
+  /** A single "invalid value" resource type. */
```

Comment:
I think this would benefit from an example to make it clearer. The thing that is potentially confusing is what level this is "invalid" at - it's not invalid at the toolkit level; it's a possible value of resource names returned from the surface, but invalid to use for actually creating or referencing a resource.